hibernate-core/src/main/java/org/hibernate/bytecode/enhance/spi/interceptor/AbstractInterceptor.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or http://www.gnu.org/licenses/lgpl-2.1.html
+ */
+package org.hibernate.bytecode.enhance.spi.interceptor;
+
+import org.hibernate.engine.spi.PersistentAttributeInterceptor;
+import org.hibernate.engine.spi.SharedSessionContractImplementor;
+
+/**
+ * @author Steve Ebersole
+ */
+public abstract class AbstractInterceptor implements PersistentAttributeInterceptor, SessionAssociableInterceptor {
+	private final String entityName;
+
+	private transient SharedSessionContractImplementor session;
+	private boolean allowLoadOutsideTransaction;
+	private String sessionFactoryUuid;
+
+	@SuppressWarnings("WeakerAccess")
+	public AbstractInterceptor(String entityName) {
+		this.entityName = entityName;
+	}
+
+	public String getEntityName() {
+		return entityName;
+	}
+
+	@Override
+	public SharedSessionContractImplementor getLinkedSession() {
+		return session;
+	}
+
+	@Override
+	public void setSession(SharedSessionContractImplementor session) {
+		this.session = session;
+		if ( session != null && !allowLoadOutsideTransaction ) {
+			this.allowLoadOutsideTransaction = session.getFactory().getSessionFactoryOptions().isInitializeLazyStateOutsideTransactionsEnabled();
+			if ( this.allowLoadOutsideTransaction ) {
+				this.sessionFactoryUuid = session.getFactory().getUuid();
```

Comment:
Is it possible that this ```AbstractInterceptor``` will become associated with a ```Session``` from a different ```SessionFactory```? If so, ```sessionFactoryUuid``` won't be updated if ```allowLoadOutsideTransaction``` == ```true```.

Reply:
Have you run into this before?  This is exactly the code as it happened before this work.  I see what you are saying, but I think that has nothing to do with this work.